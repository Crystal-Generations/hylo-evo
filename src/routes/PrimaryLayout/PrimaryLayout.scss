.container {
  display: flex;
  flex-direction: column;
  align-content: stretch;
  align-items: stretch;
  height: 100vh;
  overflow: hidden;
  &.map-view {
    height: 100vh;
  }
}

.debug {
  .main {
    border: 1px dashed red;
  }
  .left {
    border: 1px dashed blue;
  }
  .center {
    border: 1px dashed green;
  }
  .sidebar {
    border: 1px dashed magenta;
  }
  .detail {
    border: 1px dashed purple;
  }
}

.top {
  /* empty */
  margin-bottom: -1 * $space-6x;
}

.main {
  position: relative;
  flex-grow: 1;
  display: flex;
  flex-wrap: nowrap;
  width: 100%;
  max-width: 1500px;
  margin-left: auto;
  margin-right: auto;
  min-width: 700px;
  &.map-view {
    width: 100%;
    max-width: 100%;
  }
}

$drawer-width: 367px;
$drawer-height: 100vh;
$drawer-z-index: 51;
.drawer {
  width: $drawer-width;
  height: $drawer-height;
  position: fixed;
  left: 0px;
  z-index: $drawer-z-index;
  display: flex;
  flex-direction: column;
  transition-delay: 0s;
  transition-duration: 0.2s;
  &.hidden {
    left: -$drawer-width;
  }
}

.left .map-field {
    height: calc(100vh - 60px);
}

.center {
  padding-left: 3px;
  background: $color-white;
  // From Bootstrap make-col-ready():
  position: relative;
  min-height: 1px; // Prevent collapsing
  height: 100vh;
  flex: 0 0 $center-column-width;
  max-width: $center-column-width;
  overflow-y: auto;
  overflow-x: hidden;
  padding-right: $space-2x;
  padding-bottom: 100px;
  &.map-view {
    max-width: $map-center-column-width;
    flex: 0 0 $map-center-column-width;
    height: calc(100vh - 40px);
    padding-bottom: 0px;
  }
}

.center.collapsedState {
  width: 43%;
  flex: 0 0 43%;
}

.center::-webkit-scrollbar {
  background-color:transparent;
  width: 5px;
  border-radius: 5px;
}

.center::-webkit-scrollbar-thumb {
  background-color: rgba(200,200,200,.5   );
  outline: none;
  border-radius: 3px;
}

.sidebar {
  // From Bootstrap make-col-ready():
  position: relative;
  z-index: 10;
  min-height: 1px; // Prevent collapsing
  flex: 0 0 29%;
  max-width: 29%;
  transition-delay: 0.1s;
  transition-duration: 0.3s;
  right: 0px;
  padding: 0 16px 0 $layout-spacing;
  overflow-y: auto;

  &.hidden {
    opacity: 0;
    right: -60px;
    transition-delay: 0.0s;
  }
}

.detail {
  background-color: white;
  position: absolute;
  z-index: 30;
  top: 0;
  right: 0px;
  height: 100%;
  width: $detail-column-width;
  box-shadow: 0px 4px 56px 0px $shadow-color;
  transition-delay: 0.1s;
  transition-duration: 0.2s;
  padding-bottom: 150px;
  overflow-y: auto;
  margin-top: -8px;

  &.hidden {
    opacity: 0;
    right: -1000px;
    transition-delay: 0.0s;
  }
}

.map-view .detail {
  max-width: 700px;
}

.detail::-webkit-scrollbar {
  background-color:transparent;
  width: 5px;
  border-radius: 5px;
}

.detail::-webkit-scrollbar-thumb {
  background-color: rgba(200,200,200,.5);
  outline: none;
}

