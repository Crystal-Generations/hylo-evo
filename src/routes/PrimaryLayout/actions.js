import {
  FETCH_CURRENT_USER,
  TOGGLE_COMMUNITIES_DRAWER
} from 'store/constants'

export function fetchCurrentUser () {
  return {
    type: FETCH_CURRENT_USER,
    graphql: {
      query: `{
        me {
          id
          name
          avatarUrl
          memberships {
            id
            person {
              id
            }
            community {
              id
              name
              slug
              avatarUrl
            }
          }
        }
      }`
    }
  }
}

export function toggleCommunitiesDrawer () {
  return {
    type: TOGGLE_COMMUNITIES_DRAWER
  }
}
