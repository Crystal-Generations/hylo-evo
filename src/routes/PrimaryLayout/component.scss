
.content,
.detail {
  // background-color: $color-white;
  // border: dashed 1px $color-rhino-30;
}

@media (min-width: 80rem) {
  .container {
    width: 80rem;
    // @include make-container();
      position: relative;
      margin-left: auto;
      margin-right: auto;
      // @each $breakpoint in map-keys($gutters) {
      //   @include media-breakpoint-up($breakpoint) {
      //     $gutter: map-get($gutters, $breakpoint);
      //     padding-right: ($gutter / 2);
      //     padding-left:  ($gutter / 2);
      //   }
      // }
  }
}

.row {
  // @include make-row();
    display: flex;
    flex-wrap: wrap;
    // @each $breakpoint in map-keys($gutters) {
    //   @include media-breakpoint-up($breakpoint) {
    //     $gutter: map-get($gutters, $breakpoint);
    //     margin-right: ($gutter / -2);
    //     margin-left:  ($gutter / -2);
    //   }
    // }
  composes: justify-content-between from $bootstrap-path;
  // removes the negative margin from Bootrap row. There is likely a better way
  margin: 0 0 !important;
}

.header {
  // @include make-col-ready();
    position: relative;
    // Prevent columns from becoming too narrow when at smaller grid tiers by
    // always setting `width: 100%;`. This works because we use `flex` values
    // later on to override this initial width.
    width: 100%;
    min-height: 1px; // Prevent collapsing
    // @each $breakpoint in map-keys($gutters) {
    //   @include media-breakpoint-up($breakpoint) {
    //     $gutter: map-get($gutters, $breakpoint);
    //     padding-right: ($gutter / 2);
    //     padding-left:  ($gutter / 2);
    //   }
    // }
  composes: p-6 mb-6 from $bootstrap-path;
  box-shadow: 0px 3px 5px 0px $color-nepal;
  color: $color-athens-gray;
  background-color: $color-persimmon;
}

.content {
  // @include make-col-ready();
    position: relative;
    // Prevent columns from becoming too narrow when at smaller grid tiers by
    // always setting `width: 100%;`. This works because we use `flex` values
    // later on to override this initial width.
    width: 100%;
    min-height: 1px; // Prevent collapsing
    // @each $breakpoint in map-keys($gutters) {
    //   @include media-breakpoint-up($breakpoint) {
    //     $gutter: map-get($gutters, $breakpoint);
    //     padding-right: ($gutter / 2);
    //     padding-left:  ($gutter / 2);
    //   }
    // }
  // @include make-col(5.439);
  flex: 0 0 45%;
  max-width: 45%;
  overflow: auto;
  max-height: 80vh;
}

.sidebar {
  // @include make-col-ready();
    position: relative;
    // Prevent columns from becoming too narrow when at smaller grid tiers by
    // always setting `width: 100%;`. This works because we use `flex` values
    // later on to override this initial width.
    width: 100%;
    min-height: 1px; // Prevent collapsing
    // @each $breakpoint in map-keys($gutters) {
    //   @include media-breakpoint-up($breakpoint) {
    //     $gutter: map-get($gutters, $breakpoint);
    //     padding-right: ($gutter / 2);
    //     padding-left:  ($gutter / 2);
    //   }
    // }
  // @include make-col(3);
  flex: 0 0 25%;
  max-width: 25%;
  transition-delay: 0.1s;
  transition-duration: 0.3s;
  right: 0px;
  &.hidden {
    opacity: 0;
    right: -60px;
    transition-delay: 0.0s;
  }
}

.detail {
  composes: detail;
  padding-top: 24px;
  position: absolute;
  top: 56px;
  right: 0px;
  flex: 0 0 45%;
  max-width: 45%;
  box-shadow: 0px 4px 56px 0px $color-alto;
  transition-delay: 0.1s;
  transition-duration: 0.2s;
  height: 100vh;
  width: 100vw;
  &.hidden {
    opacity: 0;
    right: -580px;
    transition-delay: 0.0s;
  }
}
