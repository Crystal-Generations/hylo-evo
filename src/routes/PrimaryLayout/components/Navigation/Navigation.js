import React from 'react'
import { compact } from 'lodash/fp'
import NavLink from './NavLink'
import TopicNavigation from './TopicNavigation'
import './Navigation.scss'
import { compact } from 'lodash/fp'
import { EVENTS } from 'config/featureFlags'

export default function Navigation (props) {
  const {
    currentUser,
    className,
    collapsed,
    routeParams,
    rootPath,
    membersPath,
    projectsPath,
    communityId,
    eventsPath,
    badge,
    clearBadge,
    clearFeedList,
    hideTopics
  } = props

  const homeOnClick = () => {
    if (window.location.pathname === rootPath) {
      clearFeedList()
      clearBadge()
    }
  }

  const hasEventsFeature = currentUser && currentUser.hasFeature(EVENTS)

  const links = compact([
    rootPath && {
      label: 'Home',
      icon: 'Home',
      to: rootPath,
      badge: badge,
      onClick: homeOnClick,
      exact: true
    },
    projectsPath && {
      label: 'Projects',
      icon: 'Projects',
      to: projectsPath
    },
    eventsPath && hasEventsFeature && {
      label: 'Events',
      icon: 'Events',
      to: eventsPath
    },
    membersPath && {
      label: 'Members',
      icon: 'Members',
      to: membersPath
    }
  ])

  return <div styleName={collapsed ? 'collapser-collapsed' : 'collapser'}
    className={className}>
    <div styleName='navigation'>
      <ul styleName='links'>
        {links.map(link =>
          <NavLink key={link.label} {...link} collapsed={collapsed}
            onClick={link.onClick} />)}
      </ul>
      {!hideTopics && <TopicNavigation
        collapsed={collapsed}
        backUrl={rootPath}
        routeParams={routeParams}
        communityId={communityId} />}
    </div>
  </div>
}
