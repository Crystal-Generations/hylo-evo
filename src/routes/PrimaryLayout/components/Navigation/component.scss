@mixin make-visible {
  // background-color: $color-white;
  // border: dashed 1px $color-rhino-30;
}

.navigation {
  @include make-visible();
  @include make-col-ready();
  // @media (max-width: 32em) {
  //   @include make-col(6);
  // }
  @media (min-width: 32.1em) {
    @include make-col(2.4);
  }
  transition: 0.5s;

  // ovverriding bootstrap default. Seems broken
  padding: 0 0 !important;
  margin: 0 15px;
}

.navigation-collapsed {
  composes: navigation;

  @media (min-width: 32.1em) {
    @include make-col(0.72);
  }
}

.links {
  list-style: none;
  border-bottom: 2px solid $color-ghost-30;
  padding: 0 0 20px 0;
  margin: 0;
}
