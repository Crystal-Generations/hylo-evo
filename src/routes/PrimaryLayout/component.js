/* eslint-disable no-unused-vars */
import React, { Component } from 'react'
import { matchPath, Route, Link } from 'react-router-dom'
import SampleCard from 'components/SampleCard'
import Navigation from './components/Navigation'
import TopNav from './components/TopNav'
import cx from 'classnames'
import { get } from 'lodash/fp'
import Feed from 'routes/Feed'
import Events from 'routes/Events'
import EventDetail from 'routes/Events/EventDetail'

// Global styles
import 'css/global/index.scss'

export default function PrimaryLayout ({ match, location }) {
  const hasDetail = matchPath(location.pathname, {path: '/events/:eventId'})

  return <div styleName='container'>
    <TopNav />
    <div styleName='row'>
      <Route path='/' component={Navigation} collapsed={hasDetail} />
      <div styleName='content'>
        <Route path='/' exact component={Feed} />
        <Route path='/events' component={Events} />
      </div>
      <div styleName={cx('sidebar', {hidden: hasDetail})}>
        <Route path='/' component={Feed} />
      </div>
      <div styleName={cx('detail', {hidden: !hasDetail})}>
        // TODO: Display content of last detail page on '/' so that the
        // animation transitions correctly.
        // Best guess is to replace these routes with a render function
        // defined above, and store the previous detail component in state
        <Route path='/events/:eventId' exact component={EventDetail} />
      </div>
    </div>
  </div>
}
