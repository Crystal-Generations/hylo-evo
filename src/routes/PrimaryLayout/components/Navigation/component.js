import React from 'react'
import NavLink from './NavLink'
import TopicNavigation from './TopicNavigation'
import './component.scss'

export default function Navigation ({ location, collapsed }) {
  let links = [
    {label: 'Home', icon: 'Home', to: '/'},
    {label: 'Events', icon: 'Events', to: '/events', badge: 3},
    {label: 'Members', icon: 'Members', to: '/members'},
    {label: 'UI Kit', icon: 'Projects', to: '/ui-kit'}
  ]
  links = links.map(link => ({...link, active: (link.to === location.pathname)}))

  return <div styleName={collapsed ? 'collapser-collapsed' : 'collapser'}>
    <div styleName='navigation'>
      <ul styleName='links'>
        {links.map(link => <NavLink key={link.label} {...link} collapsed={collapsed} />)}
      </ul>
      <TopicNavigation />
    </div>
  </div>
}
