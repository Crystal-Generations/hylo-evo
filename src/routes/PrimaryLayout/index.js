import { HOLOCHAIN_ACTIVE } from 'util/holochain'
import component from './PrimaryLayout'
import holochainConnector from './PrimaryLayout.holochain.connector'
import connector from './PrimaryLayout.connector'

export default HOLOCHAIN_ACTIVE
  ? holochainConnector(component)
  : connector(component)
