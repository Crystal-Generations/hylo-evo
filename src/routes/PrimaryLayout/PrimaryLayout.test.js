import { MemoryRouter, Redirect } from 'react-router'
import React from 'react'
import { shallow } from 'enzyme'

import orm from 'store/models'
import Navigation from './components/Navigation'
import PrimaryLayout, { redirectIfCommunity, RedirectToCommunity } from './PrimaryLayout'

it('shows nothing for a community route if the community is not loaded', () => {
  const wrapper = shallow(<PrimaryLayout isCommunityRoute />)
  expect(wrapper.name()).toEqual('Loading')
})

it('shows normal children for a community route if the community is loaded', () => {
  const wrapper = shallow(<PrimaryLayout
    isCommunityRoute
    location={{pathname: '/'}}
    community={{id: '1'}} />)
  expect(wrapper.name()).toEqual('div')
})

describe('RedirectToCommunity', () => {
  let session

  beforeEach(() => {
    session = orm.mutableSession(orm.getEmptyState())
    session.Person.create({ id: '1' })
  })

  it('shows a Loading component if currentUser not set yet', () => {
    const wrapper = shallow(<RedirectToCommunity currentUser={null} />)
    const renderWrapper = shallow(wrapper.props().render())
    expect(renderWrapper.find('[data-styleName="loading-top"]').length).toBe(1)
  })

  it('shows a Loading component if currentUser has no memberships', () => {
    const user = session.Person.withId('1')
    const wrapper = shallow(redirectIfCommunity(user)())
    expect(wrapper.find('[data-styleName="loading-top"]').length).toBe(1)
  })

  it('sets `to` prop of Redirect correctly', () => {
    session.Community.create({ id: '1', slug: 'foo' })
    session.Membership.create({ id: '1', community: '1' })
    const user = session.Person.withId('1')
    user.set('memberships', [ '1' ])

    const wrapper = shallow(<MemoryRouter>
      {redirectIfCommunity(user)()}
    </MemoryRouter>)

    const expected = '/c/foo'
    const actual = wrapper.find(Redirect).props().to
    expect(actual).toBe(expected)
  })
})
