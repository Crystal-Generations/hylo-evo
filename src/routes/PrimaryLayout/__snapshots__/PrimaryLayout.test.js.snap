// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows NotFound if a currentUser is loaded and the community does not exist 1`] = `<withRouter(Connect(NotFound)) />`;

exports[`shows nothing for a community route if the community and currentUser are not loaded 1`] = `
<div
  data-stylename="container"
>
  <Loading
    type="loading-fullscreen"
  />
</div>
`;
